lib/Doctrine/DBAL/Driver/IBMDB2/DataSourceName.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Driver\IBMDB2;
+
+use function implode;
+use function sprintf;
+use function strpos;
+
+/**
+ * IBM DB2 DSN
+ */
+final class DataSourceName
+{
+    /** @var string */
+    private $string;
+
+    private function __construct(string $string)
+    {
+        $this->string = $string;
+    }
+
+    public function toString() : string
+    {
+        return $this->string;
+    }
+
+    /**
+     * Creates the object from an array representation
+     *
+     * @param array<string,mixed> $params
+     */
+    public static function fromArray(array $params) : self
+    {
+        $chunks = [];
+
+        foreach ($params as $key => $value) {
+            $chunks[] = sprintf('%s=%s', $key, $value);
```

Comment:
Should we throw or maybe trigger a deprecation if an unknown key is passed? Or do we need the door open for more params than the one appearing from lines 59 to 64?

Reply:
No. DBAL is not the source of truth about which parameters the underlying driver or the platform support.